lib/ae_mdw/stats.ex

```diff
@@ -209,6 +223,87 @@ defmodule AeMdw.Stats do
     end
   end
 
+  @spec fetch_transactions_statistics(State.t(), pagination(), query(), range(), cursor()) ::
+          {:ok, {pagination_cursor(), [statistic()], pagination_cursor()}}
+  def fetch_transactions_statistics(state, pagination, query, range, cursor) do
+    with {:ok, query} <- convert_params(query),
+         {:ok, cursor} <- deserialize_statistic_cursor(cursor) do
+      {prev_cursor, statistics_keys, next_cursor} =
+        state
+        |> build_transactions_statistics_streamer(query, range, cursor)
+        |> Collection.paginate(pagination)
+
+      statistics = Enum.map(statistics_keys, &render_statistic(state, &1))
+
+      {:ok,
+       {serialize_statistics_cursor(prev_cursor), statistics,
+        serialize_statistics_cursor(next_cursor)}}
+    end
+  end
+
+  defp build_transactions_statistics_streamer(state, query, _scope, cursor) do
+    tx_tag = Keyword.get(query, :tx_type, :all)
+
+    cursor =
+      case cursor do
+        nil -> nil
+        interval_start -> {{:transactions, tx_tag}, :day, interval_start}
+      end
+
+    key_boundary = {
+      {{:transactions, tx_tag}, :day, Util.min_int()},
+      {{:transactions, tx_tag}, :day, Util.max_int()}
+    }
+
+    fn direction ->
+      Collection.stream(state, Model.Statistic, direction, key_boundary, cursor)
+    end
+  end
+
+  defp render_statistic(state, {{:transactions, _tx_tag}, :day, interval_start} = statistic_key) do
```

Comment:
For me it's okay if you wanna add `:month` and `:year` to the same PR, but it's your choice.

Reply:
Yeah, was intending not to implement the filter yet. Also, monthly intervals would need to be rendered differently